### PR TITLE
Add list row

### PR DESCRIPTION
**Pull request by ethanjc**
This PR adds the List Row component. The list row takes a wide variety of props that allows it to be used to display icons, Logos, images, titles, and snippets of text.

https://user-images.githubusercontent.com/20915535/186281476-4b34ff5f-17e2-4412-8f12-fdc2f767fdd6.mp4

### CODE REVIEW

File: packages/constellation/src/components/Compositions/ListRow/ListRow.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import cx from 'classnames'
+import React, { ReactElement } from 'react'
+
+import { IconChevronRightV2, IconProps, PaletteColors, SemanticColors, Text } from '../../Base'
+import { Switch, Tag } from '../../Primitives'
+import { Logo } from '../index'
+import { ListRowComponent } from './ListRow.types'
+
+/**
+ * TitleByline is an internarl component that handles the title and
```
Comment (noahub) [inline packages/constellation/src/components/Compositions/ListRow/ListRow.tsx]:
```suggestion
 * TitleByline is an internal component that handles the title and
```

File: packages/constellation/src/components/Compositions/ListRow/ListRow.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        {
+          'cursor-pointer': onClick,
+        },
+        { '!items-start': hasDescription },
+      )}
+      onClick={onClick}
+    >
+      <div className={cx('flex flex-row items-center gap-4', { '!h-[47px]': hasDescription })}>
+        {icon &&
+          React.cloneElement(
```
Comment (noahub) [inline packages/constellation/src/components/Compositions/ListRow/ListRow.tsx]: Why do we have to cloneElement here?

Comment (ethanjc) [reply packages/constellation/src/components/Compositions/ListRow/ListRow.tsx]: cloneElement is just being used to set the size and colour of the icon, I suppose that could be done app side when passing the icon in though

Comment (noahub) [reply packages/constellation/src/components/Compositions/ListRow/ListRow.tsx]: Ok gotcha, I haven't seen this pattern elsewhere so was a little unsure.

**Review by blockdylanb — APPROVED** [review]

**Pull request merged**